src/WingetCreateCLI/Commands/UpdateLocaleCommand.cs

```diff
@@ -0,0 +1,291 @@
+            {
+                TelemetryManager.Log.WriteEvent(commandEvent);
+            }
+        }
+
+        private static object PromptAllLocalesAsMenuSelection(Manifests originalManifests)
+        {
+            Dictionary<string, object> localeMap = new Dictionary<string, object>
+            {
+                { originalManifests.DefaultLocaleManifest.PackageLocale + " (" + Resources.DefaultLocale_MenuItem + ")", originalManifests.DefaultLocaleManifest },
+            };
+
+            foreach (var locale in originalManifests.LocaleManifests)
+            {
+                localeMap.Add(locale.PackageLocale, locale);
+            }
+
+            string selectedLocale = Prompt.Select(Resources.SelectExistingLocale_Message, localeMap.Keys.ToList());
+            return localeMap[selectedLocale];
+        }
+
+        private static void DisplayUpdatedLocales(Manifests manifest)
+        {
+            Logger.DebugLocalized(nameof(Resources.GenerateUpdatedLocalePreview_Message));
+            if (manifest.DefaultLocaleManifest != null)
+            {
+                DisplayDefaultLocaleManifest(manifest.DefaultLocaleManifest);
+            }
+
+            if (manifest.LocaleManifests != null)
+            {
+                DisplayLocaleManifests(manifest.LocaleManifests);
+            }
+        }
+
+        private Manifests PromptAndUpdateExistingLocales(Manifests originalManifests)
+        {
+            Manifests updatedLocales = new Manifests()
+            {
+                SingletonManifest = null,
+                VersionManifest = null,
+                InstallerManifest = null,
+            };
+
+            List<string> defaultPromptPropertiesForUpdateLocale = new()
+            {
+                nameof(LocaleManifest.PackageName),
+                nameof(LocaleManifest.Publisher),
+                nameof(LocaleManifest.License),
+                nameof(LocaleManifest.ShortDescription),
+            };
+
+            bool localeArgumentUsed = false;
+            object selectedLocale;
+
+            do
+            {
+                if (!string.IsNullOrEmpty(this.Locale) && !localeArgumentUsed)
+                {
+                    selectedLocale = LocaleHelper.GetMatchingLocaleManifest(this.Locale, originalManifests);
+                    localeArgumentUsed = true;
+                }
+                else
+                {
+                    selectedLocale = PromptAllLocalesAsMenuSelection(originalManifests);
+                }
+
+                bool isDefaultLocale = false;
+
+                if (selectedLocale is LocaleManifest localeManifest)
+                {
+                    Logger.DebugLocalized(nameof(Resources.FieldSetToValue_Message), nameof(LocaleManifest.PackageLocale), localeManifest.PackageLocale);
+                    LocaleHelper.PromptAndSetLocaleProperties(localeManifest, defaultPromptPropertiesForUpdateLocale);
+                    updatedLocales.LocaleManifests.Add(localeManifest);
```

Comment:
You will probably want to add the localeManifest to `updatedLocales` once the user has completed filling out the prompts which includes the additional locale properties. That will make things below a lot simplier.

Reply:
That would require converting the DefaultLocale to a LocaleManifest first and then only adding the manifest with `updatedLocales.LocalManifests.Add()`. Given concerns with the type conversion causing us to lose properties (Moniker field), resolving https://github.com/microsoft/winget-create/pull/480#discussion_r1419714365 seems to be a prerequisite for this task.